projects/OG-Analytics/src/main/java/com/opengamma/analytics/financial/provider/calculator/discounting/CrossGammaSingleCurveCalculator.java

```diff
@@ -108,5 +120,89 @@ public DoubleMatrix2D calculateCrossGamma(final InstrumentDerivative instrument,
     DoubleMatrix2D gammaMat = new DoubleMatrix2D(gammaArray);
     return gammaMat;
   }
+  
+  /**
+   * Computes the gamma "sum-of-column" for a given instrument. See the documentation for the definition.
+   * The curve provider should contain only one curve which should be of the 
+   * type YieldCurve with an underlying InterpolatedDoublesCurve.
+   * @param instrument The instrument for which the cross-gamma should be computed.
+   * @param multicurve The multi-curve provider.
+   * @return The gamma "sum-of-columns" vector.
+   */
+  public double[] calculateSumOfColumnsGamma(final InstrumentDerivative instrument,
+      final MulticurveProviderDiscount multicurve) {
+    ArgumentChecker.isTrue(multicurve.getAllNames().size() == 1,
+        "provider should have only one curve for GammaSingleCurve computation");
+    String name = multicurve.getAllNames().iterator().next();
+    YieldAndDiscountCurve curve = multicurve.getCurve(name);
+    ArgumentChecker.isTrue(curve instanceof YieldCurve, "curve should be YieldCurve");
+    YieldCurve yieldCurve = (YieldCurve) curve;
+    ArgumentChecker.isTrue(yieldCurve.getCurve() instanceof InterpolatedDoublesCurve,
+        "Yield curve should be based on InterpolatedDoublesCurve");
+    Delta deltaShift = new Delta(multicurve, instrument, _psc);
+    Function1D<DoubleMatrix1D, DoubleMatrix2D> gammaFn = _differentiator.differentiate(deltaShift);
+    double[][] gamma2 = gammaFn.evaluate(new DoubleMatrix1D(new double[1])).getData();
+    double[] gamma = new double[gamma2.length];
+    for (int i = 0; i < gamma2.length; i++) {
+      gamma[i] = gamma2[i][0];
+    }
+    return gamma;
+  }
 
 }
+
+/**
+ * Inner class to compute the delta for a given parallel shift of the curve.
+ */
+class Delta extends Function1D<DoubleMatrix1D, DoubleMatrix1D> {  
+  private final double[] _x;
+  private final double[] _y;
+  private final int _nbNode;
+  private final String _name;
+  private final MulticurveProviderDiscount _multicurve;
+  private final InstrumentDerivative _instrument;
+  private final Currency _ccy;
+  private final InterpolatedDoublesCurve _interpolatedCurve;
+  private final ParameterSensitivityParameterCalculator<ParameterProviderInterface> _psc;
+  
+  public Delta(MulticurveProviderDiscount multicurve, InstrumentDerivative instrument, 
+      ParameterSensitivityParameterCalculator<ParameterProviderInterface> psc) {
+    _multicurve = multicurve;
+    _name = multicurve.getAllNames().iterator().next();
+    _instrument = instrument;
+    _ccy = multicurve.getCurrencyForName(_name);
+    YieldAndDiscountCurve curve = multicurve.getCurve(_name);
+    YieldCurve yieldCurve = (YieldCurve) curve;
```

Comment:
Do we really need that restriction here? Both YieldCurve and DiscountCurve have an underlying InterpolatedDoublesCurve. If we have discount factors instead of zero-rates, it should be easy enough to transform them appropriately, no?